tree/ntuple/v7/test/ntuple_print.cxx

```diff
@@ -0,0 +1,184 @@
+   EXPECT_EQ(expected, os.str());
+}
+
+TEST(RNtuplePrint, FloatPrint)
+{
+   std::stringstream os;
+   RPrintVisitor visitor(os);
+   RField<float> testField("floatTest");
+   testField.AcceptVisitor(visitor, 1);
+   std::string expected{std::string("")
+       + "********************************************************************************\n"
+       + "* Field 1   : floatTest (float)                                                *\n"};
+   EXPECT_EQ(expected, os.str());
+   
+}
+
+TEST(RNtuplePrint, FloatTraverse)
+{
+   std::stringstream os;
+   RPrintVisitor visitor(os, 'a');
+   RField<float> testField("floatTest");
+   testField.TraverseVisitor(visitor, 1);
+   std::string expected{std::string("")
+       + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\n"
+       + "a Field 1   : floatTest (float)                                                a\n"};
+   EXPECT_EQ(expected, os.str());
+}
+
+TEST(RNtuplePrint, VecAccept)
+{
+   std::stringstream os;
+   RPrintVisitor visitor(os, 'a');
+   RField<std::vector<float>> testField("floatTest");
+   testField.AcceptVisitor(visitor, 1);
+   std::string expected{std::string("")
+       + "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\n"
+       + "a Field 1   : floatTest (std::vector<float>)                                   a\n"};
+   EXPECT_EQ(expected, os.str());
+}
+
+TEST(RNtuplePrint, VecTraverse)
+{
+   std::stringstream os;
+   RPrepareVisitor prepVisitor;
+   RField<std::vector<float>> testField("floatVecTest");
+   testField.TraverseVisitor(prepVisitor, 1);
+   RPrintVisitor visitor(os, '$');
+   visitor.SetDeepestLevel(prepVisitor.GetDeepestLevel());
+   visitor.SetNumFields(prepVisitor.GetNumFields());
+   testField.TraverseVisitor(visitor, 1);
+   std::string expected{std::string("")
+       + "$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$\n"
+       + "$ Field 1       : floatVecTest (std::vector<float>)                            $\n"
+       + "$ |__Field 1    : float (float)                                                $\n"};
+   EXPECT_EQ(expected, os.str());
+}
+
+TEST(RNtuplePrint, VecVecTraverse)
+{
+   std::stringstream os;
+   RPrepareVisitor prepVisitor;
+   RField<std::vector<std::vector<float>>> testField("floatVecVecTest");
+   testField.TraverseVisitor(prepVisitor, 1);
+   RPrintVisitor visitor(os, 'x');
+   visitor.SetDeepestLevel(prepVisitor.GetDeepestLevel());
+   visitor.SetNumFields(prepVisitor.GetNumFields());
+   testField.TraverseVisitor(visitor, 1);
+   std::string expected{std::string("")
+       + "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx\n"
+       + "x Field 1           : floatVecVecTest (std::vector<std::vector<float>>)        x\n"
+       + "x |__Field 1        : std::vector<float> (std::vector<float>)                  x\n"
```

Comment:
That's a good point. In all test cases except for the first one, a stand-alone field is created instead of a ntuple-model. In such cases there is no rootfield, so the subfields of floatVecVecTest believe that floatVecVecTest is the rootfield, because it has no parent. (floatVecVecTest has the default order, which is 1) 
But it's confusing, so I'll modify RNTupleFormatter::HierarchialFieldOrder to also work for such cases by setting the default value (used by rootfield) to -1 and for the test cases manually set the order to 1. (Could also be useful later if there is a feature where only a single field is printed.)